plugins/atomicfu/atomicfu-compiler/src/org/jetbrains/kotlinx/atomicfu/compiler/backend/jvm/AtomicfuJvmIrTransformer.kt

```diff
@@ -0,0 +1,825 @@
+                        val volatileField = buildVolatileRawField(this, parentClass)
+                        backingField = volatileField
+                        getter?.transformAccessor(volatileField, getter?.dispatchReceiverParameter?.capture())
+                        setter?.transformAccessor(volatileField, setter?.dispatchReceiverParameter?.capture())
+                    } else {
+                        // Property delegated to the atomic property:
+                        // 1. delegate it's accessors to get/set of the backingField of the atomic delegate
+                        // (that is already transformed to a volatile field of value type)
+                        // val _a = atomic(0)
+                        // var a by _a ->
+                        // volatile var _a: Int = 0
+                        // var a by _a
+                        val atomicProperty = initializer.getCorrespondingProperty()
+                        val volatileField = atomicProperty.backingField!!
+                        backingField = null
+                        if (atomicProperty.isTopLevel()) {
+                            atomicSymbols.createBuilder(symbol).run {
+                                val parent = getStaticVolatileWrapperInstance(atomicProperty)
+                                getter?.transformAccessor(volatileField, getProperty(parent, null))
+                                setter?.transformAccessor(volatileField, getProperty(parent, null))
+                            }
+                        } else {
+                            if (this.parent == atomicProperty.parent) {
+                                //class A {
+                                //    val _a = atomic()
+                                //    var a by _a
+                                //}
+                                getter?.transformAccessor(volatileField, getter?.dispatchReceiverParameter?.capture())
+                                setter?.transformAccessor(volatileField, setter?.dispatchReceiverParameter?.capture())
+                            } else {
+                                //class A {
+                                //    val _a = atomic()
+                                //    inner class B {
+                                //        var a by _a
+                                //    }
+                                //}
+                                val thisReceiver = atomicProperty.parentAsClass.thisReceiver
+                                getter?.transformAccessor(volatileField, thisReceiver?.capture())
+                                setter?.transformAccessor(volatileField, thisReceiver?.capture())
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private fun IrFunction.transformAccessor(volatileField: IrField, parent: IrExpression?) {
+            val accessor = this
+            atomicSymbols.createBuilder(symbol).run {
```

Comment:
Generally, we use `with(...createBuilder(symbol))` idiom, when we create IR nodes. This helps to spot the places, where IR is create at a glance.